src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -487,9 +487,22 @@ takeWhile1 predicate (Fold fstep finitial fextract) =
 --
 -- /Internal/
 --
-sliceSepBy :: -- MonadCatch m =>
+sliceSepBy :: MonadCatch m =>
     (a -> Bool) -> Parser m a b -> Parser m a b
-sliceSepBy _cond = undefined
+sliceSepBy cond (Parser pstep pinitial pextract) =
```

Comment:
This should be named `sliceSepByP` I believe.

Reply:
It could be `sliceSepByP` though there is no `sliceSepBy` so we do not necessarily need the distinction. In some other combinators we use the P suffix where we collect the output using a Parser instead of Fold. If we want that convention to be consistent we can add `P` suffix.

Reply:
Should I rename it, then?

Reply:
Yes, we should rename it.